test/sys/test_mman.rs

```diff
@@ -43,7 +47,7 @@ fn test_mremap_grow() {
     let slice: &mut [u8] = unsafe {
         #[cfg(target_os = "linux")]
         let mem = mremap(
-            slice.as_mut_ptr().cast(),
+            NonNull::from(&slice[..]).cast(),
```

Comment:
Here you're mutating through an immutable pointer.  That's not ok.  You should do `Slice::as_mut_ptr` or `&mut slice` instead.  It happens a few places below, too.

Reply:
Updated it to `&mut slice[..]`

Reply:
It seems that we can get rid of that `#![allow(clippy::redundant_slicing)]` if we just use the `slice` here, `slice` itself is of type `&mut [u8]`, the constructed `NonNull` will be `NonNull<[u8]>`, then we `cast()` it to `NonNull<c_void>`

Reply:
I tried this, it doesn't work.

Reply:
Oops, sorry, `&mut [u8]` is not `Copy`, manually reborrowing here should do the trick